staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -18,6 +18,11 @@ package v1beta1
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+// TODO: This does not generate to protobuf because of the interface{} - fix protobuf
+// generation to support a meta type that can accept any valid JSON.
+
+// +protobuf=false
```

Comment:
protobuf generation does not work due to the presence of interface fields.

We can, however, have an intermediate representation for these fields and define a marshaller for proto on that.

We can have two cases for this intermediate representation:

1. A struct like:

```
type JSON struct {
   Float *float
   String *string
   Map map[string]JSON
   ...
}
```

2. A string.

@smarterclayton wdyt?